dali/aux/optical_flow/turing_of/optical_flow_turing.cu

```diff
@@ -0,0 +1,108 @@
+// Copyright (c) 2017-2018, NVIDIA CORPORATION. All rights reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#include "dali/aux/optical_flow/turing_of/optical_flow_turing.h"
+#include "dali/error_handling.h"
+
+namespace dali {
+namespace optical_flow {
+namespace kernel {
+
+namespace {
+
+constexpr size_t kBlockSize = 32;
+constexpr int kBgrBytesPerPixel = 3;
+constexpr int kAbgrBytesPerPixel = 4;
+constexpr int kBytesPerEncodedFlowComponent = sizeof(int16_t);
+constexpr int kBytesPerDecodedFlowComponent = sizeof(float);
+
+
+/**
+ * Calculating number of blocks
+ * @param length In bytes
+ * @param block_size
+ * @return
+ */
+inline size_t num_blocks(size_t length, size_t block_size) {
+  // Calculating ceil for ints
+  return (length + block_size - 1) / block_size;
+}
+
+/**
+ * Calculate value in strided buffer according to (x, y) coordinates
+ * @param buffer
+ * @param x In pixels
+ * @param y In pixels
+ * @param pitch Stride, in bytes
+ * @return Value at given coordinates
+ */
+template<typename T>
+__host__ __device__ constexpr T &pitch_xy(T *buffer, ptrdiff_t x, ptrdiff_t y, ptrdiff_t pitch) {
```

Comment:
```suggestion
__host__ __device__ constexpr T &pitch_xy(T *buffer, ptrdiff_t x, ptrdiff_t y, ptrdiff_t pitch_bytes) {
```
I think we should be very explicit about stride being in _bytes_, not in elements of type `T`

Reply:
done